addons/test_mail/tests/test_mail_activity.py

```diff
@@ -692,6 +692,49 @@ def test_my_activity_flow_employee(self):
         with self.with_user('employee'):
             record = self.env['mail.test.activity'].search([('my_activity_date_deadline', '=', date_today)])
             self.assertEqual(test_record_1, record)
+        
+
+    def test_activity_read_group_raw(self):
+        Activity = self.env['mail.activity']
+        date_today = date.today()
+        Activity.create({
+            'activity_type_id': self.env.ref('test_mail.mail_act_test_todo').id,
+            'date_deadline': date_today,
+            'res_model_id': self.env.ref('test_mail.model_mail_test_activity').id,
+            'res_id': self.test_record.id,
+            'user_id': self.user_admin.id,
+        })
+        Activity.create({
+            'activity_type_id': self.env.ref('test_mail.mail_act_test_call').id,
+            'date_deadline': date_today ,
+            'res_model_id': self.env.ref('test_mail.model_mail_test_activity').id,
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+        })
+        Activity.create({
+            'activity_type_id': self.env.ref('test_mail.mail_act_test_todo').id,
+            'date_deadline': date_today,
+            'res_model_id': self.env.ref('test_mail.model_mail_test_activity').id,
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+        })
+        Activity.create({
+            'activity_type_id': self.env.ref('test_mail.mail_act_test_todo').id,
+            'date_deadline': date_today + relativedelta(days=7),
+            'res_model_id': self.env.ref('test_mail.model_mail_test_activity').id,
+            'res_id': self.test_record.id,
+            'user_id': self.user_employee.id,
+        })
+        groupby = "date_deadline"
+        domain = [(1,"=",1)]
+        groups = Activity.with_user(self.user_employee)._read_group_raw(domain ,
+                                        fields = ['activity_type_id','date_deadline','user_id'] ,
+                                        groupby=[groupby],
+                                        )
+        for i in groups:
+            read_result2 = Activity.search(i.get('__domain'))
+        self.assertTrue(read_result2)
```

Comment:
I'm not sure exactly what we're trying to test here. We only test the last result, I assume that's not inteded?

Globally I think the test should just check that `_read_group_raw` applies the same rules as read_group and such. There's an example for the regular read_group at `test_activity_security_user_noaccess_manual` near `# cannot read_group activities if no access to the document`.

Just adding a test for `_read_group_raw` in one of those methods would be fine too :)

Hope that clears it up a bit